subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/TaskCachingHashesListener.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.execution;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import org.gradle.api.Task;
+import org.gradle.caching.BuildCacheKey;
+
+import java.util.Map;
+import java.util.Set;
+
+public interface TaskCachingHashesListener {
+    void inputsCollected(Task task, BuildCacheKey key, TaskCachingInputs hashes);
```

Comment:
nit, I like the order:

`inputsCollected(BuildCacheKey key, TaskCachingInputs hashes, Task task)`

Because it flows a bit like, "inputs collected for `key` are `hashes` from `task`"